docs/hugo/content/contributing/create-a-new-release.md

```diff
@@ -17,21 +22,88 @@ title: Creating a new release
-6. Wait for it to start: `k get all -n azureserviceoperator-system`
-7. Create a resource group and a vnet in it (the vnet is to check that conversion webhooks are working, since there aren't any for RGs):
+
+Perform a simple smoke test to make sure the new release is capable of starting up and creating some Azure resources.
+
+1. Create a kind cluster
+   ``` bash
+   task controller:kind-create
    ```
-   k apply -f v2/samples/resources/v1beta/v1beta20200601_resourcegroup.yaml
-   k apply -f v2/samples/network/v1beta/v1beta20201101_virtualnetwork.yaml
+
+2. Install cert-manager
+   ``` bash
+   task controller:install-cert-manager
+   ```
+
+3. Create the namespace for the operator
+   ``` bash
+   kubectl create namespace azureserviceoperator-system
+   ```
+
+4. Ensure you have a [Service Principal](https://azure.github.io/azure-service-operator/#installation) and export these environment variables:
+
+   * AZURE_SUBSCRIPTION
+   * AZURE_TENANT_ID
+   * AZURE_CLIENT_ID
+   * AZURE_CLIENT_SECRET
+
+5. Create a secret in your cluster with the Service Principal credentials for ASO to use:
+   ``` bash
+   task controller:make-sp-secret
+   ```
+
+6. Download the operator from MCR.
+
+   ``` bash 
+   wget https://github.com/Azure/azure-service-operator/releases/download/v2.3.0/azureserviceoperator_v2.3.0.yaml
+
+7. Configure [`crd-patterns`](https://azure.github.io/azure-service-operator/guide/crd-management/) to use with new release.
+
+   ``` bash
+   sed -i 's_--crd-pattern=.*_--crd-pattern=resources.azure.com/*;network.azure.com/*_g' azureserviceoperator_v2.3.0.yaml
+   ```
+
+   We use `sed` to replace the blank setting included in the release with the one we want to use, including both resource group and network CRDs.
+
+8. Install the operator into your cluster
+   
+   ``` bash
+   kubectl apply --server-side=true -f azureserviceoperator_v2.0.0-beta.3.yaml
+   ```
+
+   (We need to use [server-side apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) because the CRD for VirtualMachines is large enough that it can't fit in the `last-applied-configuration` annotation client-side `kubectl apply` uses.)
+
+7. Watch while ASO starts
+   ``` bash
+   kubectl get all -n azureserviceoperator-system
+   ```
+
+8. Create a resource group and a vnet in it (the vnet is to check that conversion webhooks are working, since there aren't any for RGs):
+
+   ``` bash
+   kubectl apply -f v2/samples/resources/v1api/v1api20200601_resourcegroup.yaml
+   kubectl apply -f v2/samples/network/v1api20201101/v1api20201101_virtualnetwork.yaml
    ```
 8. Make sure they deploy successfully - check in the portal as well.
 
```

Comment:
Can you also fix up the "Creating and testing Helm chart for new release" to refer to CRDpattern in the install step?
I think it just needs the default CRD pattern applied... maybe with networking too?

Reply:
Done.